## Cho tập tin "shark.tsv" chứa dữ liệu số vụ cá mập tấn công các tháng trong năm của 3 quốc gia USA, Australia, South Africa)

In [2]:
import numpy as np

In [19]:
#1. Đọc dữ liệu từ tập tin shark.tsv và đưa vào mảng shark(Gợi ý: sử dụng hàm np.genfromtxt) 
number_shark_attacks = np.genfromtxt(fname = 'shark.tsv', delimiter='\t', dtype = "int")
number_shark_attacks = number_shark_attacks[1:]
number_shark_attacks

array([[ -1,  28,  94,  68],
       [ -1,  27,  78,  32],
       [ -1,  66,  63,  34],
       [ -1, 103,  54,  25],
       [ -1, 106,  21,  25],
       [ -1, 173,  25,  22],
       [ -1, 232,  25,  28],
       [ -1, 208,  28,  15],
       [ -1, 182,  35,  17],
       [ -1, 141,  56,  16],
       [ -1,  79,  65,  20],
       [ -1,  37,  94,  46]])

<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
  
  <pre>array([[ -1,  28,  94,  68],
       [ -1,  27,  78,  32],
       [ -1,  66,  63,  34],
       [ -1, 103,  54,  25],
       [ -1, 106,  21,  25],
       [ -1, 173,  25,  22],
       [ -1, 232,  25,  28],
       [ -1, 208,  28,  15],
       [ -1, 182,  35,  17],
       [ -1, 141,  56,  16],
       [ -1,  79,  65,  20],
       [ -1,  37,  94,  46]])
</pre>

</details>

In [20]:
#2. Cho biết số phần tử mỗi chiều của mảng
number_shark_attacks.shape

(12, 4)

<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
  
  <pre>(12, 4)
</pre>

</details>

In [22]:
#3. Thay cột 1 (giá trị -1) bằng giá trị tương ứng các tháng trong năm 1-12
number_shark_attacks[:, 0] = np.arange(1, number_shark_attacks.shape[0]+1)
number_shark_attacks

array([[  1,  28,  94,  68],
       [  2,  27,  78,  32],
       [  3,  66,  63,  34],
       [  4, 103,  54,  25],
       [  5, 106,  21,  25],
       [  6, 173,  25,  22],
       [  7, 232,  25,  28],
       [  8, 208,  28,  15],
       [  9, 182,  35,  17],
       [ 10, 141,  56,  16],
       [ 11,  79,  65,  20],
       [ 12,  37,  94,  46]])

<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
  
  <pre>array([[  1,  28,  94,  68],
       [  2,  27,  78,  32],
       [  3,  66,  63,  34],
       [  4, 103,  54,  25],
       [  5, 106,  21,  25],
       [  6, 173,  25,  22],
       [  7, 232,  25,  28],
       [  8, 208,  28,  15],
       [  9, 182,  35,  17],
       [ 10, 141,  56,  16],
       [ 11,  79,  65,  20],
       [ 12,  37,  94,  46]])
</pre>

</details>

In [32]:
#4. Hiển thị số vụ cá mập tấn công trung bình của 3 quốc gia
# sum_USA =np.array([np.sum(number_shark_attacks[:, i])/12 for i in range(1,number_shark_attacks.shape[1]+1)]) 
sum_USA = np.array([np.sum(number_shark_attacks[:, i]) / 12 for i in range(1,number_shark_attacks.shape[1])])  
sum_USA

array([115.16666667,  53.16666667,  29.        ])

<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
  
  <pre>array([115.16666667,  53.16666667,  29.        ])
</pre>

</details>

In [30]:
#5. Hiển thị số vụ cá mập tấn công trung bình của mỗi tháng  
sum_month = np.array([np.sum(number_shark_attacks[i, 1:]) / (number_shark_attacks.shape[1]-1) for i in range(number_shark_attacks.shape[0])])  

sum_month

array([63.33333333, 45.66666667, 54.33333333, 60.66666667, 50.66666667,
       73.33333333, 95.        , 83.66666667, 78.        , 71.        ,
       54.66666667, 59.        ])

<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
  
  <pre>array([63.33333333, 45.66666667, 54.33333333, 60.66666667, 50.66666667,
       73.33333333, 95.        , 83.66666667, 78.        , 71.        ,
       54.66666667, 59.        ])
</pre>

</details>

In [35]:
#6. Tạo thêm cột tổng số vụ cá mập tấn công theo mỗi tháng 
sum_month_shark_attacks = np.array([np.sum(number_shark_attacks[i, 1:]) for i in range(number_shark_attacks.shape[0])])  

number_shark_attacks = np.insert(number_shark_attacks, number_shark_attacks.shape[1], values=sum_month_shark_attacks, axis=1)
number_shark_attacks 


array([[  1,  28,  94,  68, 190],
       [  2,  27,  78,  32, 137],
       [  3,  66,  63,  34, 163],
       [  4, 103,  54,  25, 182],
       [  5, 106,  21,  25, 152],
       [  6, 173,  25,  22, 220],
       [  7, 232,  25,  28, 285],
       [  8, 208,  28,  15, 251],
       [  9, 182,  35,  17, 234],
       [ 10, 141,  56,  16, 213],
       [ 11,  79,  65,  20, 164],
       [ 12,  37,  94,  46, 177]])

<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
  
  <pre>array([[  1,  28,  94,  68, 190],
       [  2,  27,  78,  32, 137],
       [  3,  66,  63,  34, 163],
       [  4, 103,  54,  25, 182],
       [  5, 106,  21,  25, 152],
       [  6, 173,  25,  22, 220],
       [  7, 232,  25,  28, 285],
       [  8, 208,  28,  15, 251],
       [  9, 182,  35,  17, 234],
       [ 10, 141,  56,  16, 213],
       [ 11,  79,  65,  20, 164],
       [ 12,  37,  94,  46, 177]])
</pre>

</details>

In [8]:
#7. Cho biết những tháng có tổng số vụ cá mập tấn công >200


array([[  6, 173,  25,  22, 220],
       [  7, 232,  25,  28, 285],
       [  8, 208,  28,  15, 251],
       [  9, 182,  35,  17, 234],
       [ 10, 141,  56,  16, 213]])

<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
  
  <pre>array([[  6, 173,  25,  22, 220],
       [  7, 232,  25,  28, 285],
       [  8, 208,  28,  15, 251],
       [  9, 182,  35,  17, 234],
       [ 10, 141,  56,  16, 213]])
</pre>

</details>

In [9]:
#8. Cho biết tổng số vụ cá mập tấn công nhiều nhất trong tháng là bao nhiêu


285

<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
  
  <pre>285
</pre>

</details>

In [10]:
#9. Cho biết tháng nào có tổng số vụ cá mập tấn công nhiều nhất


Tháng có tổng số vụ cá mập tấn công nhiều nhất là tháng : 7


<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
  
  <pre>Tháng có tổng số vụ cá mập tấn công nhiều nhất là tháng : 7
</pre>

</details>